src/stdlib_linalg_least_squares.fypp

```diff
@@ -203,12 +203,18 @@ submodule (stdlib_linalg) stdlib_linalg_least_squares
          !! Local variables
          type(linalg_state_type) :: err0
          integer(ilp) :: m,n,lda,ldb,nrhs,ldx,nrhsx,info,mnmin,mnmax,arank,lrwork,liwork,lcwork
-         integer(ilp) :: nrs,nis,ncs,nsvd
+         integer(ilp) :: nrs,nis,nsvd
+         #:if rt.startswith('complex')         
+         integer(ilp) :: ncs
+         #:endif
          integer(ilp), pointer :: iwork(:)
          logical(lk) :: copy_a,large_enough_x
          real(${rk}$) :: acond,rcond
          real(${rk}$), pointer :: rwork(:),singular(:)
-         ${rt}$, pointer :: xmat(:,:),amat(:,:),cwork(:)
+         ${rt}$, pointer :: xmat(:,:),amat(:,:)
+         #:if rt.startswith('complex')         
```

Comment:
Is this procedure `startwith` supported by all versions of Python?

Reply:
@jvdp1 I think @cyrilgandon is just following the same style that I've used through the BLAS/LAPACK routines for those cases that have differences between `real` and `complex` versions. As far as I can tell, string's `startwith` was supported in Python 2 already. Another way to accomplish the same result would be to use as used elsewhere:

https://github.com/fortran-lang/stdlib/blob/de9b1f8f6cb5482c897f93e06062243b4c635e01/src/stdlib_io.fypp#L138

Reply:
Thank you for the explanation @perazz . As you can see, my knowledge in Python is quite limited ;) If this approach (using `startwith`) is already used and works, I am fine with it!